sycl/include/CL/sycl/detail/pi.h

```diff
@@ -805,6 +805,21 @@ __SYCL_EXPORT pi_result piPlatformGetInfo(pi_platform platform,
                                           void *param_value,
                                           size_t *param_value_size_ret);
 
+/// Gets the native handle of a PI platform object.
+///
+/// \param platform is the PI platform to get the native handle of.
+/// \param nativeHandle is the native handle of platform.
+__SYCL_EXPORT pi_result piextPlatformGetNativeHandle(
+    pi_platform platform, pi_native_handle *nativeHandle);
+
+/// Creates PI platform object from a native handle.
+/// NOTE: The created PI object takes ownership of the native handle.
```

Comment:
Maybe we should clarify what "takes ownership" actually means in terms of retaining / releasing the native object.

Reply:
this is a bigger task to [re-]define "ownership" tracked here: https://github.com/intel/llvm/issues/1516